Mathlib/Algebra/BigOperators/Fin.lean

```diff
@@ -306,6 +308,15 @@ end PartialProd
 
 end Fin
 
+namespace Finsupp
+
+lemma sum_cons {M} [AddCommMonoid M] (n : ℕ) (σ : Fin n →₀ M) (i : M) :
+    (sum (Finsupp.cons i σ) fun _ e ↦ e) = i + sum σ (fun _ e ↦ e) := by
```

Comment:
I think this is more generally true as
```suggestion
lemma sum_cons {M} [AddCommMonoid M] (n : ℕ) (σ : Fin n →₀ M) (i : M)
  (h : f 0 0 = 0) :
    (sum (Finsupp.cons i σ) f) = f 0 i + sum σ f := by
```

Reply:
Hmm, it would be nice if it were analogous to the Fin.sum_cons def though. Is it ok if I do both and rename one?

Reply:
Sure, I think having a primed one for the more complex version is fine.

Reply:
Ok I added the primed one, it looks a little different than yours - I changed h to what lean said it needed. let me know if I did it wrong.